server/neptune/workflows/internal/terraform/runners/job_runner.go

```diff
@@ -0,0 +1,98 @@
+package runners
+
+import (
+	"strings"
+
+	"github.com/hashicorp/go-version"
+	"github.com/runatlantis/atlantis/server/neptune/workflows/internal/github"
+	"github.com/runatlantis/atlantis/server/neptune/workflows/internal/steps"
+	"go.temporal.io/sdk/workflow"
+)
+
+// CustomRunner runs custom run steps.
+type StepRunner interface {
+	Run(
+		ctx workflow.Context,
+		step steps.Step,
+		repo github.Repo,
+		commit github.Commit,
+		tfVersion *version.Version,
+		projectName string,
+		repoRelDir string,
+		path string,
+		envs map[string]string,
+	) (string, error)
+}
+
+// JobRunner runs a deploy job
+type JobRunner interface {
+	Run(
+		ctx workflow.Context,
+		job steps.Job,
+		repo github.Repo,
+		commit github.Commit,
+		tfVersion *version.Version,
+		projectName string,
+		repoRelDir string,
+		path string,
+	) (string, error)
+}
+
+func NewJobRunner(
+	runStepRunner StepRunner,
+	envStepRunner StepRunner,
+) JobRunner {
```

Comment:
Go best practice is to always return a concrete implementation in these construction functions.

Reply:
Gotcha, will keep that in mind :)